centos_ci.sh

```diff
@@ -208,27 +208,103 @@ function docs_tar_upload() {
   echo "Find docs tar here http://artifacts.ci.centos.org/minishift/minishift/docs/$LATEST."
 }
 
-if [[ "$UID" = 0 ]]; then
+function create_release_commit() {
+  # Set Terminal
+  export TERM=xterm-256color
+  # Add git a/c identity
+  git config user.email "minishift@lists.minishift.io"
+  git config user.name "Minishift Bot"
+  # Export GITHUB_ACCESS_TOKEN
+  export GITHUB_ACCESS_TOKEN=$GITHUB_TOKEN
+  # Create master branch as git clone in CI doesn't create it
+  git checkout -b master
+  # Bump version and commit
+  sed -i "s|MINISHIFT_VERSION = .*|MINISHIFT_VERSION = $RELEASE_VERSION|" Makefile
+  git add Makefile
+  git commit -m "cut v$RELEASE_VERSION"
+  # Push update
+  git push https://$REPO_OWNER:$GITHUB_TOKEN@github.com/$REPO_OWNER/minishift master
+}
+
+function add_release_notes() {
+  release_id=$(curl -s "https://api.github.com/repos/${REPO_OWNER}/minishift/releases" | jq --arg release "v$RELEASE_VERSION" -r ' .[] | if .name == $release then .id else empty end')
+
+  if [[ "$release_id" != "" ]]; then
+    MILESTONE_ID=`curl -s https://api.github.com/repos/minishift/minishift/milestones | jq --arg version "v$RELEASE_VERSION" -r ' .[] | if .title == $version then .number else empty end'`
+    # Generate required json payload for release note
+    ./scripts/release/issue-list.sh -r minishift -m $MILESTONE_ID | jq -Rn 'inputs + "\n"' | jq -s '{body:  add }' > json_payload.json
+    # Add release notes
+    curl -H "Content-Type: application/json" -H "Authorization: token $GITHUB_TOKEN" \
+         --data @json_payload.json https://api.github.com/repos/${REPO_OWNER}/minishift/releases/$release_id
+
+    echo "Release notes of Minishift v$RELEASE_VERSION has been successfully updated. Find the release notes here https://github.com/${REPO_OWNER}/minishift/releases/tag/v$RELEASE_VERSION."
+  else
+    echo "Failed to update release notes of Minishift v$RELEASE_VERSION as couldn't find the release ID. Try to manually update the release notes here https://github.com/${REPO_OWNER}/minishift/releases/tag/v$RELEASE_VERSION."
+    exit 1
+  fi
+}
+
+
+function setup_build_environment() {
   load_jenkins_vars;
   prepare_ci_user;
   install_core_deps;
   install_kvm_virt;
   install_docker;
   runuser -l minishift_ci -c "/bin/bash centos_ci.sh"
+}
+
+function perform_release() {
+  cd gopath/src/github.com/minishift/minishift
+  create_release_commit;
+
+  make release
+
+  if [[ "$?" = 0 ]]; then
+    echo "Minishift v$RELEASE_VERSION has been successfully released. Find the latest release here https://github.com/$REPO_OWNER/minishift/releases/tag/v$RELEASE_VERSION."
+  else
+    echo "Failed to release Minishift v$RELEASE_VERSION. Try to release manually."
+    exit 1
+  fi
+
+  add_release_notes;
+  # Generate docs tar and upload
+  make gen_adoc_tar IMAGE_UID=$(id -u)
+  # Handle error on failure of doc tar generation
+  if [[ "$?" != 0 ]]; then
```

Comment:
@hferentschik I think this check is not necessary. If there is any failure in above step, whole build process will abort like https://ci.centos.org/job/minishift-release/58/console